source/common/init/manager_impl.cc

```diff
@@ -51,8 +61,9 @@ void ManagerImpl::initialize(const Watcher& watcher) {
 
     // Attempt to initialize each target. If a target is unavailable, treat it as though it
     // completed immediately.
-    for (const auto& target_handle : target_handles_) {
-      if (!target_handle->initialize(watcher_)) {
+    for (auto iter = target_handles_.begin(); iter != target_handles_.end();
```

Comment:
I was told before not to do this (although I honestly don't remember why). In any case, I'd say that the simplest and safest way to clear the list of target handles is to just call `target_handles_.clear()` after traversing it.

Reply:
I am curious about the scenario you were told not to. My wild guess is the timing/threadcontext. Please let me know when it comes to your mind.
Meanwhile I will change to `clear()` for this PR